File: app/src/main/java/com/krystianwsul/checkme/persistencemodel/UuidRecord.java
package com.krystianwsul.checkme.persistencemodel;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.support.annotation.NonNull;
import android.text.TextUtils;

import junit.framework.Assert;

import java.util.UUID;

class UuidRecord extends Record {
    static final String TABLE_UUID = "uuid";

    private static final String COLUMN_UUID = "uuid";

    @NonNull
    private final String mUuid;

    @NonNull
    static String newUuid() {
        String uuid = UUID.randomUUID().toString();
        Assert.assertTrue(!TextUtils.isEmpty(uuid));

        return uuid;
    }

    public static void onCreate(@NonNull SQLiteDatabase sqLiteDatabase) {
        sqLiteDatabase.execSQL("CREATE TABLE " + TABLE_UUID
                + " (" + COLUMN_UUID + " TEXT NOT NULL);");

        UuidRecord uuidRecord = new UuidRecord(false, newUuid());
        uuidRecord.getInsertCommand().execute(sqLiteDatabase);
    }

    @NonNull
    static UuidRecord getUuidRecord(@NonNull SQLiteDatabase sqLiteDatabase) {
        Cursor cursor = sqLiteDatabase.query(TABLE_UUID, null, null, null, null, null, null);
        cursor.moveToFirst();

        UuidRecord uuidRecord = cursorToCustomTimeRecord(cursor);

        Assert.assertTrue(cursor.isLast());

        return uuidRecord;
    }

    @NonNull
    private static UuidRecord cursorToCustomTimeRecord(@NonNull Cursor cursor) {
        String uuid = cursor.getString(0);
        Assert.assertTrue(!TextUtils.isEmpty(uuid));

        return new UuidRecord(true, uuid);
    }

    UuidRecord(boolean created, @NonNull String uuid) {
        super(created);

        Assert.assertTrue(!TextUtils.isEmpty(uuid));

        mUuid = uuid;
    }

    @NonNull
    String getUuid() {
        return mUuid;
    }

    @NonNull
    @Override
    ContentValues getContentValues() {
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUMN_UUID, mUuid);

        return contentValues;
    }

    @NonNull
    @Override
    UpdateCommand getUpdateCommand() {
        throw new UnsupportedOperationException();
    }

    @NonNull
    @Override
    InsertCommand getInsertCommand() {
        return getInsertCommand(TABLE_UUID);
    }

    @NonNull
    @Override
    DeleteCommand getDeleteCommand() {
        throw new UnsupportedOperationException();
    }
}

File: app/src/main/java/com/krystianwsul/checkme/persistencemodel/UuidRecord.kt
package com.krystianwsul.checkme.persistencemodel

import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase

import java.util.*

class UuidRecord(created: Boolean, val uuid: String) : Record(created) {

    companion object {

        const val TABLE_UUID = "uuid"

        private const val COLUMN_UUID = "uuid"

        fun newUuid() = UUID.randomUUID().toString()

        fun onCreate(sqLiteDatabase: SQLiteDatabase) {
            sqLiteDatabase.execSQL("CREATE TABLE $TABLE_UUID " +
                    "($COLUMN_UUID TEXT NOT NULL);")

            UuidRecord(false, newUuid()).insertCommand.execute(sqLiteDatabase)
        }

        fun getUuidRecord(sqLiteDatabase: SQLiteDatabase) = getRecords(sqLiteDatabase, TABLE_UUID, this::cursorToCustomTimeRecord).single()

        private fun cursorToCustomTimeRecord(cursor: Cursor): UuidRecord {
            val uuid = cursor.getString(0)
            check(uuid.isNotEmpty())

            return UuidRecord(true, uuid)
        }
    }

    init {
        check(uuid.isNotEmpty())
    }

    override val contentValues
        get() = ContentValues().apply {
        put(COLUMN_UUID, uuid)
    }

    override val commandTable = TABLE_UUID
    override val commandIdColumn get() = throw UnsupportedOperationException()
    override val commandId get() = throw UnsupportedOperationException()
}
